tests/functional/test_install.py

```diff
@@ -281,11 +281,12 @@ def test_install_editable_uninstalls_existing_from_path(script, data):
 
 @need_mercurial
 def test_basic_install_editable_from_hg(script, tmpdir):
-    """Test cloning from Mercurial."""
+    """Test cloning and hg+file install from Mercurial."""
     pkg_path = _create_test_package(script, name='testpackage', vcs='hg')
     args = ['install', '-e', 'hg+%s#egg=testpackage' % path_to_url(pkg_path)]
     result = script.pip(*args, **{"expect_error": True})
     result.assert_installed('testpackage', with_files=['.hg'])
+    assert path_to_url(pkg_path).startswith("file://")
```

Comment:
I would save this into a variable and use it in the construction of `args` and in the `assert`. It may also make more sense to move the `assert` above the `args` declaration. Normally we assert as close as possible to the place where something could've failed.